src/test/rbd_mirror/test_ImageReplayer.cc

```diff
@@ -119,34 +119,39 @@ class TestImageReplayer : public TestFixture {
     uint64_t snap_id;
     EXPECT_EQ(0, m_remote_ioctx.selfmanaged_snap_create(&snap_id));
 
+    m_pool_meta_cache.set_local_pool_meta(
+      m_local_ioctx.get_id(), {m_local_mirror_uuid});
+
     uint64_t features = FEATURES;
     if (MIRROR_IMAGE_MODE == cls::rbd::MIRROR_IMAGE_MODE_JOURNAL) {
       EXPECT_EQ(0, librbd::api::Mirror<>::mode_set(m_remote_ioctx,
                                                    RBD_MIRROR_MODE_POOL));
       EXPECT_EQ(0, librbd::api::Mirror<>::mode_set(m_local_ioctx,
                                                    RBD_MIRROR_MODE_POOL));
+      m_pool_meta_cache.set_remote_pool_meta(
+        m_remote_ioctx.get_id(), "peer uuid",
+        {m_remote_mirror_uuid, ""});
     } else {
       EXPECT_EQ(0, librbd::api::Mirror<>::mode_set(m_remote_ioctx,
                                                    RBD_MIRROR_MODE_IMAGE));
       EXPECT_EQ(0, librbd::api::Mirror<>::mode_set(m_local_ioctx,
                                                    RBD_MIRROR_MODE_IMAGE));
 
-
       uuid_d uuid_gen;
       uuid_gen.generate_random();
       std::string remote_peer_uuid = uuid_gen.to_string();
 
+      m_pool_meta_cache.set_remote_pool_meta(
+        m_remote_ioctx.get_id(), "peer uuid",
+        {m_remote_mirror_uuid, remote_peer_uuid});
+
       EXPECT_EQ(0, librbd::cls_client::mirror_peer_add(
         &m_remote_ioctx, {remote_peer_uuid,
                           cls::rbd::MIRROR_PEER_DIRECTION_RX_TX,
                           "siteA", "client", m_local_mirror_uuid}));
 
-      m_pool_meta_cache.set_remote_pool_meta(
-        m_remote_ioctx.get_id(), {m_remote_mirror_uuid, remote_peer_uuid});
     }
 
-    EXPECT_EQ(0, librbd::api::Mirror<>::uuid_get(m_remote_ioctx,
-                                                 &m_remote_mirror_uuid));
     EXPECT_EQ(0, librbd::api::Mirror<>::uuid_get(m_local_ioctx,
                                                  &m_local_mirror_uuid));
```

Comment:
The handling of mirror uuid/mirror peer uuid seems to be really messy/hacky right now. The tests pass now but I will try to figure why I need to do this to make it works (and hopefully have something eventually better).